extensions/database/tests/src/com/google/refine/extension/database/DatabaseModuleImplTest.java

```diff
@@ -0,0 +1,95 @@
+
+import static org.testng.Assert.assertEquals;
+
+import java.util.Properties;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class DatabaseModuleImplTest {
+
+    private Properties originalProperties;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        // Save the original properties
+        originalProperties = DatabaseModuleImpl.extensionProperties;
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        // Restore the original properties
+        DatabaseModuleImpl.extensionProperties = originalProperties;
+    }
+
+    @DataProvider(name = "batchSizeTestData")
+    public Object[][] provideBatchSizeTestData() {
+        return new Object[][] {
+                // propertyName, propertyValue, methodToTest, expectedValue, message
+                // Valid value tests
+                { "create.batchSize", "500", "create", 500, "Should return the correct create batch size" },
+                { "preview.batchSize", "200", "preview", 200, "Should return the correct preview batch size" },
+                // Invalid value tests
+                { "create.batchSize", "not-a-number", "create", DatabaseModuleImpl.DEFAULT_CREATE_BATCH_SIZE,
+                        "Should return default create batch size for invalid input" },
+                { "preview.batchSize", "not-a-number", "preview", DatabaseModuleImpl.DEFAULT_PREVIEW_BATCH_SIZE,
+                        "Should return default preview batch size for invalid input" },
+                // Invalid value tests - Decimal value tests
+                { "create.batchSize", "2.5", "create", DatabaseModuleImpl.DEFAULT_CREATE_BATCH_SIZE,
+                        "Should return default create batch size when value is decimal" },
+                { "preview.batchSize", "2.5", "preview", DatabaseModuleImpl.DEFAULT_PREVIEW_BATCH_SIZE,
+                        "Should return default preview batch size when value is decimal" },
+                // Value less than one tests
+                { "create.batchSize", "-5", "create", DatabaseModuleImpl.DEFAULT_CREATE_BATCH_SIZE,
+                        "Should return default create batch size when value is less than 1" },
+                { "preview.batchSize", "-5", "preview", DatabaseModuleImpl.DEFAULT_PREVIEW_BATCH_SIZE,
+                        "Should return default preview batch size when value is less than 1" },
+        };
+    }
+
+    @Test(dataProvider = "batchSizeTestData")
+    public void testBatchSizes(String propertyName, String propertyValue, String methodType,
+            int expectedValue, String message) throws Exception {
+        Properties testProps = new Properties();
+        testProps.setProperty(propertyName, propertyValue);
+        DatabaseModuleImpl.extensionProperties = testProps;
+
+        int result;
+        if ("create".equals(methodType)) {
```

Comment:
parameter `methodType` is not needed as it is always the same as the prefix of `propertyName`
--> simply use 
`if ("create.batchSize".equals(propertyName)) { ...` 
and remove parameter `methodType` from method